### PR TITLE
[extend] Add naming best practices for schema

### PR DESCRIPTION
**Pull request by paultyng**

### CODE REVIEW

**Comment by paultyng** [comment]
@appilon good call, I split it to a new page.

File: content/source/docs/extend/best-practices/index.html.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ practice is also linked in navigation on the left.
 
 The Best Practices section is a work in progress, with more sections to come.
 
+## Naming
+
+Naming resources, data sources, and attributes in plugins is how plugin authors
```
Comment (appilon) [inline content/source/docs/extend/best-practices/index.html.md]: maybe link to the page on the first word "Naming".

File: content/source/docs/extend/best-practices/naming.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
+layout: "extend"
+page_title: "Extending Terraform: Best Practices"
+sidebar_current: "docs-extend-best-practices-naming"
```
Comment (appilon) [inline content/source/docs/extend/best-practices/naming.md]: We just need a sidebar entry, you can add it here:
https://github.com/hashicorp/terraform-website/blob/master/content/source/layouts/extend.erb#L45

Also super minor nitpick, it looks like so far the entries and paragraphs on the index page are in alphabetical order. I don't mind having naming be first though, I actually prefer the content flows from "beginner stuff --> advanced stuff", but I will leave it up to you I really don't find that matters.

Comment (paultyng) [reply content/source/docs/extend/best-practices/naming.md]: Yeah I thought naming was kind of a foundational one, so I thought first made sense.

**Comment by paultyng** [comment]
@appilon ok, I think I addressed the additional feedback.

**Pull request merged**